proxy/upgrade.go

```diff
@@ -108,6 +125,13 @@ func (p *upgradeProxy) serveHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	if resp.StatusCode != http.StatusSwitchingProtocols {
+		log.Errorf("Got invalid status code from backend: %d", resp.StatusCode)
```

Comment:
we don't control the backend behavior, we should log this only as Debug(), because it is not our error

Reply:
but then we should also not error on Auth check or maybe just delete the `http.StatusUnauthorized` check just above this if check?

Reply:
i think we should remove logging an error on unauthorized. It's not an error and we return the right status code.